include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -1934,7 +1937,19 @@ bool cpp_dec_float<Digits10, ExponentType, Allocator>::rd_string(const char* con
       if (((pos = str.find('e')) != std::string::npos) || ((pos = str.find('E')) != std::string::npos))
       {
          // Remove the exponent part from the string.
+         #ifndef BOOST_MP_STANDALONE
          exp = boost::lexical_cast<exponent_type>(static_cast<const char*>(str.c_str() + (pos + 1u)));
+         #else
+         BOOST_IF_CONSTEXPR(std::is_integral<exponent_type>::value)
+         {
+            exp = static_cast<exponent_type>(std::atoll(static_cast<const char*>(str.c_str() + (pos + 1u))));
+         }
+         else
+         {
+            static_assert(sizeof(exponent_type) == 1, "Can not use this exponent type in standalone mode. Please de-activate and try again");
```

Comment:
Hi Matt. I believe in this particular change in line 1949, you might mean

```C
static_assert(sizeof(exponent_type) > 1, "...");
```

Reply:
This is designed to fail if the else branch is hit.